machine/os/detect.go

```diff
@@ -0,0 +1,5 @@
+package os
```

Comment:
As mentioned in IRC: Let's rename the package to `provision` since `os` is a common Go stdlib package.